### PR TITLE
Rules matching perf improvements

### PR DESCRIPTION
**Pull request by mstemm**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

> /kind release

> If contributing rules or changes to rules, please make sure to also uncomment one of the following line:

> /kind rule-update

> /kind rule-create

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

/area engine

> /area rules

> /area tests

> /area proposals

> /area CI

**What this PR does / why we need it**:

Some small changes to rules matching that reduce cpu usage with high event volumes (> 1M syscalls/sec)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Some small changes to rules matching that reduce cpu usage with high event volumes (> 1M syscalls/sec)
```

### CODE REVIEW

File: userspace/engine/falco_engine.cpp [hunk 1]
```diff
@@ -367,7 +379,15 @@ std::size_t falco_engine::add_source(const std::string &source,
 	// evttype_index_ruleset is the default ruleset implementation
 	std::shared_ptr<filter_ruleset_factory> ruleset_factory(
 		new evttype_index_ruleset_factory(filter_factory));
-	return add_source(source, filter_factory, formatter_factory, ruleset_factory);
+	size_t idx = add_source(source, filter_factory, formatter_factory, ruleset_factory);
+
+	if(source == "syscall")
```
Comment (leogr) [inline userspace/engine/falco_engine.cpp:384]:
```suggestion
	if(source == falco_common::syscall_source)
```

Comment (mstemm) [reply userspace/engine/falco_engine.cpp]: Oh yes! Fixed.

**Review by jasondellaluce — APPROVED** [review]
/approve

**Pull request merged**